src/sentry/api/endpoints/release_file_details.py

```diff
@@ -104,6 +113,10 @@ def get(self, request, project, version, file_id):
         except ReleaseFile.DoesNotExist:
             raise ResourceDoesNotExist
 
+        if request.GET.get('download') is not None and (
+           request.access.has_scope('project:releases') or
```

Comment:
We should stick with just `project:write`.

Checking for `project:releases` is kinda moot at this point anyways since literally any member has `project:releases` permission. But as of now, `project:releases` doesn't have any concept of `read` vs `write`, so this is effectively opening it up to everybody.

We should limit to `project:write` and document this behavior. We can always loosen constraint here if we have the need in the future. I'm hesitant on exposing this feature to people who potentially didn't have access to source code before, so I feel `project:write` is sufficient to start.